source/Octostache/Templates/Functions/TextManipulationFunction.cs

```diff
@@ -95,5 +96,61 @@ public static string Trim(string argument, string[] options)
                     return null;
             }
         }
+
+        public static string UriPart(string argument, string[] options)
+        {
+            if (argument == null)
+                return null;
+
+            if (!options.Any())
+                return $"[{nameof(UriPart)}: no argument given]";
+
+            if (!Uri.TryCreate(argument, UriKind.RelativeOrAbsolute, out var uri))
+                return argument;
```

Comment:
The tests are comprehensive! However, I noticed the tests miss the case where the URI cannot be parsed, and the original URI is returned as-is. Consider adding that scenario.

Reply:
Uri.TryCreate == false is a hard case to test. If you manage to find an input where that method returns false, then tell me about it! :)

Reply:
In that case! Happy to leave it out.

A tip worth considering: Does decompiling and inspecting the code give you any hints on when `TryCreate` might fail? _I can't decompile myself right now._

Reply:
The source is available from [Microsoft](https://referencesource.microsoft.com/#system/net/system/URI.cs). It seems that parsing errors are handled for `UriKind.Absolute`. When using `UriKind.RelativeOrAbsolute` everything gets past.

Reply:
I won't pursue this any further.